File: internal/sem/testdata/fixtures/java-oo/Animals.java
package zoo;

interface Named {
    String name();
}

interface Loud extends Named {
    String shout();
}

abstract class Animal implements Named {
    public String name() {
        return "animal";
    }
}

public class Dog extends Animal implements Loud {
    public String shout() {
        return "woof";
    }
}

File: internal/sem/testdata/fixtures/python-oo/animals.py
from typing import overload


class Named:
    def name(self):
        return "thing"


class Loud:
    def shout(self):
        return "noise"


class Dog(Named, Loud):
    def shout(self):
        return "woof"


# @overload stubs are type-only and must NOT be emitted as symbols; only the
# implementation `describe` below should appear.
@overload
def describe(x: int) -> str: ...
@overload
def describe(x: str) -> str: ...
def describe(x):
    return str(x)
